### ISSUE TITLE
CookieJar.extract_cookies doesn't process cookies form local domains when domain is explicitly set in header

### ISSUE THREAD

**Issue by d66cf4e6-8ba1-4fca-a38e-e5c696ea7295** [python/cpython#90233]
BPO | [46075](https://bugs.python.org/issue46075)
--- | :---
Nosy | @keddad
PRs | <li>python/cpython#30108</li>

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = None
created_at = <Date 2021-12-14.21:08:16.652>
labels = ['type-bug', 'library', '3.10', '3.11']
title = "CookieJar.extract_cookies doesn't process cookies form local domains when domain is explicitly set in header"
updated_at = <Date 2021-12-14.21:12:37.790>
user = 'https://github.com/keddad'
```

bugs.python.org fields:
```python
activity = <Date 2021-12-14.21:12:37.790>
actor = 'keddad'
assignee = 'none'
closed = False
closed_date = None
closer = None
components = ['Library (Lib)']
creation = <Date 2021-12-14.21:08:16.652>
creator = 'keddad'
dependencies = []
files = []
hgrepos = []
issue_num = 46075
keywords = ['patch']
message_count = 1.0
messages = ['408564']
nosy_count = 1.0
nosy_names = ['keddad']
pr_nums = ['30108']
priority = 'normal'
resolution = None
stage = 'patch review'
status = 'open'
superseder = None
type = 'behavior'
url = 'https://bugs.python.org/issue46075'
versions = ['Python 3.10', 'Python 3.11']
```

</p></details>

**Comment by JelleZijlstra** [comment]
Fixed in #30108 for 3.11. Thanks @keddad for your contributions!

I'm not planning to backport the change to 3.10, so I'm closing this issue.

**Comment by 5er9e1** [comment]
> Fixed in https://github.com/python/cpython/pull/30108 for 3.11

@JelleZijlstra 
Could you help me with the problem? You said that the problem is fixed for `3.11`, but I still can see it in `3.11.10`. I described details [here](https://github.com/psf/requests/issues/6835). Is it expected behavior to replace `localhost` with `.localhost`? As I know modern browsers and any http clients doesn't add leading dot. More over leading dot can not be used when we are using `samesite` attribute with `Lax` and `Strict` values. Is this problem planned to be fixed? Because of the issue it is not possible to use local environment for cookie based web tests and development.
